File: sca-web/src/dtos/StudyClassDto.ts
import { ProfessorDto } from "./ProfessotDto";
import { StudentDto } from "./StudentDto";
import { SubjectDto } from "./SubjectDto";

export interface StudyClassDto {
    id: number;
    code: string;
    name: string;
    classroom: string;
    description: string;
    startTime: string;

    subject?: SubjectDto;
    professor?: ProfessorDto;
    students?: StudentDto[];
}

File: sca-server/src/main/java/com/sca/sistemaControleAcademico/DTOS/StudyClass/StudyClassDto.java
package com.sca.sistemaControleAcademico.DTOS.StudyClass;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class StudyClassDto {
    private Integer id;
    private String code;
    private String name;
    private String classroom;
    private String description;
    private String startTime;
}